coffeeshop/src/main/java/ro/eduarddudu/coffeeshop/ShopFinder.kt

```diff
@@ -0,0 +1,61 @@
+package ro.eduarddudu.coffeeshop
+
+import android.location.Location
+import ro.eduarddudu.coffeeshop.Objects.Shop
+import java.util.*
+
+object ShopFinder {
+
+    @Throws(Exception::class)
+    fun findClosestLocations(latitude: Double, longitude: Double, path: String): List<Shop> {
+
+        val shops = CSVReader.parseCSV(path)
+                ?: throw Exception("Input file can not be parsed.")
+
+        if (shops.size == 0) {
+            throw Exception("Invalid input file.")
+        }
+
+        if (shops.size == 1) {
+            return shops
+        } else {
+            val userLocation = Location("userLocation")
```

Comment:
The `Location` class is in the android package, this means that you have a dependency on the android platform and your code won't run outside it. 

Usually is a good idea to keep your business logic and platform independent as possible, do you happen to know why?

Reply:
I know are another ways to calculate the distance between 2 cardinal points. But this was my choose in that moment. I understand is better to don`t have the dependency on a specific platform for the reason you can use in other languages projects.